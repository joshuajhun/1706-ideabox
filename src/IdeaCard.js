import React from 'react';

const IdeaCard = ({ title, description, id, deleteIdea }) => (
  <div>
    <h1>{title}</h1>
    <p>{description}</p>
    <button onClick={ () => deleteIdea(id) } >
      Delete
    </button>
  </div>
)

export default IdeaCard;
