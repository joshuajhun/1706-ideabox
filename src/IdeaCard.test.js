import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import IdeaCard from './ideaCard'

describe('IdeaCard', () => {
  it('should match the IdeaCard snapshot', () => {
    const renderedComponent = shallow(<IdeaCard title="changed"
                                                description='programming is hard' />)

    expect(renderedComponent).toMatchSnapshot()
  })

  it('should have a different title', () => {
    const renderedComponent = shallow(<IdeaCard title="diff title"
                                                description='programming is hard' />)

    expect(renderedComponent).toMatchSnapshot()
  })
});
