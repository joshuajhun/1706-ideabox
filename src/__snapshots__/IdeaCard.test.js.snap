// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard should have a different title 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaCard
    description="programming is hard"
    title="diff title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          diff title
</h1>,
        <p>
          programming is hard
</p>,
        <button
          onClick={[Function]}
>
          Delete
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "diff title",
        },
        "ref": null,
        "rendered": "diff title",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "programming is hard",
        },
        "ref": null,
        "rendered": "programming is hard",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            diff title
</h1>,
          <p>
            programming is hard
</p>,
          <button
            onClick={[Function]}
>
            Delete
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "diff title",
          },
          "ref": null,
          "rendered": "diff title",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "programming is hard",
          },
          "ref": null,
          "rendered": "programming is hard",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`IdeaCard should match the IdeaCard snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaCard
    description="programming is hard"
    title="changed"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          changed
</h1>,
        <p>
          programming is hard
</p>,
        <button
          onClick={[Function]}
>
          Delete
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "changed",
        },
        "ref": null,
        "rendered": "changed",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "programming is hard",
        },
        "ref": null,
        "rendered": "programming is hard",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            changed
</h1>,
          <p>
            programming is hard
</p>,
          <button
            onClick={[Function]}
>
            Delete
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "changed",
          },
          "ref": null,
          "rendered": "changed",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "programming is hard",
          },
          "ref": null,
          "rendered": "programming is hard",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
